src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1539,6 +1539,25 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
 
     const clang::Decl &dcl = static_cast<const clang::Decl &>(*decl.getDecl());
 
+    // pull in the type that is used to qualify the decl. This is needed so that
+    // `C<0>::f()` where `C<0>` is a template specialization and `f` is a static member function
+    // works correctly when `C<0>::f` is the only use of `C<0>`.
+    // When parsing the template `C` we do not parse the specializations, so there
+    // is no other code that could parse the type of `C<0>`.
+    // (Yes, clang allows us to get all specializations of a template, but that leads to other problems.
+    // See #1782 or #2284)
+
+    if (const auto nns = decl.getQualifier())
+    {
+      if (const auto type = nns->getAsType())
+      {
+        assert(!nns->isDependent());
```

Comment:
Shall we consider adding a debug log to identify cases where `isDependent()` is triggered?

Reply:
This is intended to be a sanity check, I don't think having a nested name specifier that is dependent is actually possible in this overall context, but you can never be too sure.